src/init_interp/initInterpMultilevelContainer.F90

```diff
@@ -576,14 +576,24 @@ subroutine get_levmaxurbgrnd_metadata(ncid_source, ncid_dest, &
        ! source and dest soil profiles differ despite having the same number of
        ! layers. The model will handle such cases without scaling
        ! the finidat variables that need scaling by soil layer thickness.
-       allocate(dzsoi_data_source(beg_dest:end_dest, nlev_source))
        do level = 1, nlev_source
           do g = beg_dest, end_dest
              dzsoi_data_source(g,level) = dzsoi_data_dest(g,level)
           end do
        end do
-    else  ! fail with helpful error message
-       call interp_levgrnd_check_source_file(ncid_source, dzsoi_varname)
+    else  ! different number of layers and .not. on_source
+       ! Rather than failing when DZSOI is not present on the finidat file,
```

Comment:
- [ ] Can you please add a comment making it clear that this is just for backwards compatibility (as with the else if block above)? This really helps us figure out later what code can be removed. I prefer a comment with initials & date like this `! BACKWARDS_COMPATIBILITY(wjs, 2021-04-22)` to give some indication of when we might be able to remove this backwards compatibility code.